assets/src/edit-story/components/panels/textStyle/font.js

```diff
@@ -71,10 +71,25 @@ function FontControls({ selectedElements, pushUpdate }) {
             ariaLabel={__('Font family', 'web-stories')}
             options={fonts}
             value={fontFamily}
-            onChange={(value) => {
+            onChange={async (value) => {
               const fontObj = fonts.find((item) => item.value === value);
+              const newFont = {
+                family: value,
+                ...objectPick(fontObj, [
+                  'service',
+                  'fallbacks',
+                  'weights',
+                  'styles',
+                  'variants',
+                ]),
+              };
               const { weights } = fontObj;
-
+              await maybeEnqueueFontStyle(
```

Comment:
Shouldn't `newFontWeight` from few lines down be here as well? Also the previously set font style should be here as well, no?

Reply:
> Also the previously set font style should be here as well, no?

Didn't get this, I mean, we are not invoking `fontStyle` on this component. Are you suggesting to insert this value request as a explicit `const fontStyle = getCommonValue(selectedElements, 'fontStyle');`?

Reply:
This will change a lot with `content`, which was making it very hard to direct the pull request. But that's now done. So here's what should happen: no matter what changes, you will always need to do the same thing: get new props (e.g. fontFamily), extract all styles from the `content`, and ask `maybeEnqueue` with all of them.

E.g. if you have `newFontFamily = Comic` and the content is `<b>hello <i>world</i></b> and <i>rejoice!</i>`. You need to load the following font combinations:

1. `{family: Comic, fontWeight: bold}`. I.e. you construct `font: bold 20px Comic`.
2.`{family: Comic, fontWeight: bold, fontStyle: italic}`. E.g. `font: bold italic 20px Comic`.
3.`{family: Comic, fontStyle: italic}`. E.g. `font: bold italic 20px Comic`.

There's really no way to save on anything here: all three variants here could be served by different font files. And when you change again from "Comic" to "Other" font, you'd have to repeat 1-3 with new font family.

The good news: the 1-3 parallel loading is already done by you. So I think you just need to take care of the context->style extraction.